src/connector/src/parser/postgres.rs

```diff
@@ -464,3 +520,110 @@ fn pg_numeric_to_string(val: Option<PgNumeric>) -> Option<String> {
         None
     }
 }
+
+#[derive(Clone, Debug)]
+struct EnumParser {
+    value: String,
+}
+
+impl<'a> FromSql<'a> for EnumParser {
+    fn from_sql(
+        _ty: &Type,
+        raw: &'a [u8],
+    ) -> Result<Self, Box<dyn std::error::Error + 'static + Sync + Send>> {
+        Ok(EnumParser {
+            value: String::from_utf8_lossy(raw).into_owned(),
+        })
+    }
+
+    fn accepts(ty: &Type) -> bool {
+        matches!(ty.kind(), Kind::Enum(_))
+    }
+}
+
+impl ToSql for EnumParser {
+    to_sql_checked!();
+
+    fn accepts(ty: &Type) -> bool {
+        matches!(ty.kind(), Kind::Enum(_))
+    }
+
+    fn to_sql(
+        &self,
+        ty: &Type,
+        out: &mut BytesMut,
+    ) -> Result<IsNull, Box<dyn std::error::Error + Sync + Send>>
+    where
+        Self: Sized,
+    {
+        match ty.kind() {
+            Kind::Enum(e) => {
+                if e.contains(&self.value) {
+                    out.extend_from_slice(self.value.as_bytes());
+                    return Ok(IsNull::No);
+                }
+            }
+            _ => {
+                return Err("EnumParser can only be used with ENUM types".into());
+            }
+        }
+        out.extend_from_slice(self.value.as_bytes());
```

Comment:
Shall we remove this line? If `ty.kind()` is not enum, just throw an error.

Reply:
Updated